source/common/common/envoy_defaults.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include "source/common/singleton/threadsafe_singleton.h"
+
+namespace Envoy {
+
+struct DefaultsProfile {
```

Comment:
Maybe more of a general question: this design seems to require each profile to define values for all fields, or otherwise compilation will break. It might be useful that future profiles will extend other profiles (e.g., the default profile), and only overwrite a few of the fields values.

Reply:
If a value is not defined in the manifest the macro will use the `default_value` parameter.